cli/connhelper/connhelper.go

```diff
@@ -53,3 +56,19 @@ func GetCommandConnectionHelper(cmd string, flags ...string) (*ConnectionHelper,
 		Host: "http://docker",
 	}, nil
 }
+
+func multiplexingArgs() []string {
+	if v := os.Getenv("DOCKER_SSH_NO_MUX"); v != "" {
+		if b, err := strconv.ParseBool(v); err == nil && b {
+			return nil
+		}
+	}
+	if err := os.MkdirAll(config.Dir(), 0700); err != nil {
```

Comment:
Probably we should create a new directory, because when `~/.docker` already exists with other permission bits, we can't guarantee the sockets are created under a directory with 0700. 
On my Macintosh, `~/.docker` was somehow created with 0755.

But it is more likely to hit `UNIX_PATH_MAX` (108)... :cry:

Reply:
Probably we can let `commandconn` change working directory for avoiding 108 limit

Reply:
The socket itself is created with `0600`. This mkdir is to avoid the case where `.docker` does not exist.